pydantic_ai_slim/pydantic_ai/tools.py

```diff
@@ -45,6 +51,11 @@ class RunContext(Generic[AgentDeps]):
     tool_name: str | None
     """Name of the tool being called."""
 
+    def end_run(self, result: ResultData) -> NoReturn:
+        """End the call to `agent.run` as soon as possible, using the provided value as the result."""
+        # NOTE: this means we ignore any other tools called concurrently
```

Comment:
mention in docs what happens about cancelling other tools. Do the return values of functions that finish get added to messages?

Reply:
> mention in docs what happens about cancelling other tools.

I added this comment to match the comments near the result schema things:

```python
        elif model_response.role == 'model-structured-response':
            if self._result_schema is not None:
                # if there's a result schema, and any of the calls match one of its tools, return the result
                # NOTE: this means we ignore any other tools called here
                if match := self._result_schema.find_tool(model_response):
```

Is there a discussion of this in the docs? If so I can adapt that; if not, I guess we could add a note about this too?

Reply:
I would propose that we wait to update the docs about this until after we merge Jeremiah's PR that adds the "eager" vs. "correct" mode, and support reflect those modes here too